source/SIL.AppBuilder.Portal/src/routes/(authenticated)/tasks/[product_id]/+page.server.ts

```diff
@@ -61,47 +98,107 @@ export const load = (async ({ params, url, locals }) => {
+          ProductBuild: {
+            BuildId: product?.WorkflowBuildId
+          },
+          //filter by artifact type
+          ArtifactType:
+            typeof snap.context.includeArtifacts === 'string'
+              ? snap.context.includeArtifacts
+              : undefined //include all
+        },
+        select: {
+          ProductBuildId: true,
+          ContentType: true,
+          FileSize: true,
+          Url: true,
+          Id: true
+        }
+      })
+    : [];
 
   return {
-    actions: [],
-    taskTitle: "Waiting",
-    instructions: "waiting",
+    actions: snap
+      ? filterTransitions(
+          DefaultWorkflow.getStateNodeById(`${DefaultWorkflow.id}.${snap.value}`).on,
+          snap.context
+        )
+          .filter((a) => {
+            if (session?.user.userId === undefined) return false;
+            switch (a[0].meta?.user) {
+              case RoleId.AppBuilder:
+                return session.user.userId === product?.Project.Owner.Id;
+              case RoleId.Author:
+                return product?.Project.Authors.map((a) => a.UserId).includes(session.user.userId);
+              case RoleId.OrgAdmin:
+                return product?.Project.Organization.UserRoles.map((u) => u.UserId).includes(
+                  session.user.userId
+                );
+              default:
+                return false;
+            }
+          })
+          .map((a) => a[0].eventType as string)
+      : [],
+    taskTitle: snap?.value,
+    instructions: snap?.context.instructions,
     //filter fields/files/reviewers based on task once workflows are implemented
     //possibly filter in the original query to increase database efficiency
     fields: {
-      ownerName: product?.Project.Owner.Name,
-      ownerEmail: product?.Project.Owner.Email,
+      ...filterObject(
+        {
+          ownerName: product?.Project.Owner.Name,
+          ownerEmail: product?.Project.Owner.Email,
+          storeDescription: product?.Store?.Description,
+          listingLanguageCode: product?.StoreLanguage?.Name,
+          projectURL: product?.Project.WorkflowAppProjectUrl,
+          productDescription: product?.ProductDefinition.Name,
+          appType: product?.ProductDefinition.ApplicationTypes.Description,
+          projectLanguageCode: product?.Project.Language
+        },
+        ([k, v]) => {
+          return snap?.context.includeFields.includes(k) ?? false;
+        }
+      ),
       projectName: product?.Project.Name,
-      projectDescription: product?.Project.Description,
-      storeDescription: product?.Store?.Description,
-      listingLanguageCode: product?.StoreLanguage?.Name,
-      projectURL: product?.Project.WorkflowAppProjectUrl,
-      productDescription: product?.ProductDefinition.Name,
-      appType: product?.ProductDefinition.ApplicationTypes.Description,
-      projectLanguageCode: product?.Project.Language
+      projectDescription: product?.Project.Description
     } as Fields,
```

Comment:
Is this the only place you're using your filterObject function? Often if you're using a filterObject function it means something could be done differently. I'm not entirely opposed to having it, but think about it...